cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -189,7 +192,9 @@ public RunRecord apply(RunRecordMeta input) {
   private final MetricStore metricStore;
   private final MRJobInfoFetcher mrJobInfoFetcher;
   private final NamespaceQueryAdmin namespaceQueryAdmin;
-  protected final Scheduler programScheduler;
+  private final Scheduler programScheduler;
+  private final AuthenticationContext authenticationContext;
+  private final AuthorizationEnforcer authorizationEnforcer;
```

Comment:
Authorization should happen in `Scheduler`, right? In all other cases we authorize in the service, not in the handler